bleak/backends/bluezdbus/signals.py

```diff
@@ -1,50 +1,211 @@
 # -*- coding: utf-8 -*-
 
 import asyncio
+import re
+from typing import Any, Coroutine, Dict, Optional
 
-from bleak.backends.bluezdbus.defs import PROPERTIES_INTERFACE, OBJECT_MANAGER_INTERFACE
+from dbus_next.aio.message_bus import MessageBus
+from dbus_next.errors import InvalidObjectPathError
+from dbus_next.message import Message
+from dbus_next.validators import (
+    assert_interface_name_valid,
+    assert_member_name_valid,
+    assert_object_path_valid,
+)
 
+from bleak.backends.bluezdbus.defs import PROPERTIES_INTERFACE
 
-def listen_properties_changed(bus, callback):
-    """Create a future for a PropertiesChanged signal listener.
+# TODO: this stuff should be improved and submitted upstream to dbus-next
+# https://github.com/altdesktop/python-dbus-next/issues/53
 
-    Args:
-        bus: The system bus object to use.
-        callback: The callback function to run when signal is received.
+_message_types = ["signal", "method_call", "method_return", "error"]
+
+
+class InvalidMessageTypeError(TypeError):
```

Comment:
I think I put this here instead since it is part of the code that should be upstreamed eventually. It is more like a part of `dbus-next` rather than a part of `Bleak`.